### PR TITLE
Add initial notifications OSD-6364

### PR DESCRIPTION
**Pull request by aweiteka**
Obsoletes https://github.com/openshift/ops-sop/pull/1224

### CODE REVIEW

**Comment by aweiteka** [comment]
@mmazur @RiRa12621 please review

**Comment by aweiteka** [comment]
`osdctl` servicelog implemented in [version v0.4.0](https://github.com/openshift/osdctl/releases/tag/v0.4.0)

File: README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Select the corresopnding message and send as a `POST` to 
+[api.openshift.com servicelog](https://api.openshift.com/?urls.primaryName=Service%20logs#/default/post_api_service_logs_v1_cluster_logs). 
+`CLUSTER_UUID` is a required parameter.
+
+### Examples
+
+osdctl
+
+    osdctl servicelog post -t <notificationTemplateUrl> -p CLUSTER_UUID=<UUID>
```
Comment (RiRa12621) [inline README.md:15]: this needs a cross link to where `osdctl` can be found, how to use and how to authenticate

File: osd/StuckNewBuilds3MinSRE.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "severity": "Error",
+    "service_name": "SREManualAction",
+    "cluster_uuid": "${CLUSTER_UUID}",
+    "summary": "Action required: update cluster build configuration",
+    "description": "cluster requires you to take action. Builds are failing and must be resolved. Please check the build error logs and update your configuration. For more information refer to https://docs.openshift.com/dedicated/4/builds/understanding-image-builds.html",
```
Comment (RiRa12621) [inline osd/StuckNewBuilds3MinSRE.json:6]: what builds?

File: osd/KubePersistentVolumeFillingUpWarn.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "severity": "Warning",
+    "service_name": "SREManualAction",
+    "cluster_uuid": "${CLUSTER_UUID}",
+    "summary": "Action required: update cluster logging configuration",
+    "description": "cluster requires you to take action. Elasticsearch is close to capacity. SRE will increase capacity one time causing a period while Elasticsearch is unavailable. Old logs will be retained. Logs for pods deleted while migration is in progress may be lost. SRE recommends reducing application logging on your cluster to ensure logging continues to function smoothly in the future.",
```
Comment (RiRa12621) [inline osd/KubePersistentVolumeFillingUpWarn.json:6]: needs info to which capacity

File: osd/KubePersistentVolumeFillingUpError.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+    "severity": "Error",
+    "service_name": "SREManualAction",
+    "cluster_uuid": "${CLUSTER_UUID}",
+    "summary": "Urgent action required: update cluster logging configuration",
+    "description": "cluster requires you to take action. Elasticsearch is close to capacity. Red Hat SRE strongly recommends immediately reducing application logging on your cluster to ensure logging continues to function in the near future. If logging stack stops functioning the logging stack will need reprovisioned to recover and application logs will be lost.",
```
Comment (RiRa12621) [inline osd/KubePersistentVolumeFillingUpError.json:6]: needs info to which capacity

**Comment by RiRa12621** [comment]
Made some comments.
nothing major, just clarifications.
I wonder if the messages should start with "The cluster" or "your cluster" just so it reads a little nicer

**Comment by wgordon17** [comment]
@RiRa12621 

> Made some comments.
> nothing major, just clarifications.
> I wonder if the messages should start with "The cluster" or "your cluster" just so it reads a little nicer

It feeds into an email template that starts with "Your" and then has a parameter for the description field.

It may be worth removing that, so we have a little more flexibility in the message content that SREP can post

**Comment by mmazur** [comment]
> It may be worth removing that, so we have a little more flexibility in the message content that SREP can post

I vote we drop that from the email templates. It's awkward to have to reconcile the message as it shows up in OCM vs how it shows up in the email. Templates should give full control of the messaging.

**Comment by mmazur** [comment]
(Otherwise lgtm. I'd like to start using this asap and then work on refining the messages.)

File: README.md (hunk 1 above)
Comment (aweiteka) [reply README.md:15]: done

File: osd/cluster_has_gone_missing.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+    "severity": "Error",
+    "service_name": "SREManualAction",
+    "cluster_uuid": "${CLUSTER_UUID}",
+    "summary": "Action required: delete unregistered cluster",
+    "description": "cluster requires you to take action because it is no longer registering with Red Hat OpenShift Cluster Manager. Please delete this cluster.",
```
[the remark this answers is not in the document]
Comment (aweiteka) [reply osd/cluster_has_gone_missing.json:6]: added more guidance

File: osd/StuckNewBuilds3MinSRE.json (hunk 2 above)
Comment (aweiteka) [reply osd/StuckNewBuilds3MinSRE.json:6]: image builds. updated to make this more clear

File: osd/KubePersistentVolumeFillingUpWarn.json (hunk 3 above)
Comment (aweiteka) [reply osd/KubePersistentVolumeFillingUpWarn.json:6]: added percent capacity param. fortunately osdctl will not post with undefined params

File: osd/KubePersistentVolumeFillingUpError.json (hunk 4 above)
Comment (aweiteka) [reply osd/KubePersistentVolumeFillingUpError.json:6]: added percent load

**Comment by aweiteka** [comment]
Let's defer msg formatting for later when we can coordinate with corresponding email template changes.

**Pull request merged**